### PR TITLE
Fix potential infinite recursion when increasing the pool

### PR DESCRIPTION
**Pull request by blasten**
Fixes https://github.com/PolymerElements/iron-list/issues/162

### CODE REVIEW

File: iron-list.html [hunk 1]
```diff
@@ -756,10 +756,12 @@
      * if the physical size is shorter than `_optPhysicalSize`
      */
     _increasePoolIfNeeded: function() {
-      if (this._viewportSize !== 0 && this._physicalSize < this._optPhysicalSize) {
-        // 0 <= `currentPage` <= `_maxPages`
-        var currentPage = Math.floor(this._physicalSize / this._viewportSize);
+      if (this._viewportSize !== 0 &&
+          this._physicalSize < this._optPhysicalSize &&
+          this._virtualEnd < this._virtualCount-1) {
```
Comment (blasten) [inline iron-list.html]: This issue happens when the `_physicalSize` is smaller than `_optPhysicalSize` and the user calls `scrollToIndex` to one of last indexes. 

`scrollToIndex` does `this._virtualStart = idx - 1;` and that would result in `_physicalSize` never reaching `_optPhysicalSize`.

 https://github.com/PolymerElements/iron-list/blob/master/iron-list.html#L1155

**Comment by blasten** [comment]
@ebidel would you mind testing this branch in Chrome Platform status?

**Comment by ebidel** [comment]
@blasten This fixes the CPU issue! However, when I deep link to a feature at the bottom of the list, everything above it doesn't render/rerender...even when scrolling up.

screen shot 2016-01-12 at 10 37 44 am

**Comment by ebidel** [comment]
I take that back. Looks like a weird problem in Canary.

**Comment by blasten** [comment]
@ebidel Thanks!  Did you update chromestatus.com?

**Comment by ebidel** [comment]
No, but I can stage it for you.

**Comment by ebidel** [comment]
Example: https://2016-01-12-dot-cr-status.appspot.com/features/5711871906676736

**Comment by blasten** [comment]
great!

**Comment by blasten** [comment]
@ebidel btw the last issue was fixed by https://github.com/PolymerElements/iron-list/pull/144 😅

**Comment by blasten** [comment]
I'm closing this PR as I can't reproduce the issue anymore with the latest fix in master.

**Pull request closed without merging**